From af86c85f4971fd7eb652c64d962f946368a83bcc Mon Sep 17 00:00:00 2001
From: Vince Perri <viperri@microsoft.com>
Date: Tue, 8 Feb 2022 00:16:38 +0000
Subject: [PATCH] Remove !PREEMPT_RT depends for MEMCG

MEMCG is a cgroup needed for linx containerization, but the PREEMPT_RT patch
disables it as early as v5.14-rc3-rt1. Optimizations made in the MEMCG module
made it incompatible with the real-time patch. See:

    https://lore.kernel.org/all/Ybo3cAkyg0SrUyJJ@dhcp22.suse.cz/T/

There doesn't seem to be much risk in enabling this despite the concern. MEMCG
option was added back in v5.16-rc6-rt12:

    https://lore.kernel.org/all/?q=ANNOUNCE+PREEMPT_RT+CONFIG_MEMCG

So when Mariner 2.0's LTS kernel catches up, this patch will no longer be
necessary.

---
 init/Kconfig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/init/Kconfig b/init/Kconfig
index 0b8a65ae1..45bcaa8e7 100644
--- a/init/Kconfig
+++ b/init/Kconfig
@@ -938,7 +938,6 @@ config PAGE_COUNTER
 
 config MEMCG
 	bool "Memory controller"
-	depends on !PREEMPT_RT
 	select PAGE_COUNTER
 	select EVENTFD
 	help
-- 
2.17.1

